token_bucket_test.go

```diff
@@ -135,3 +137,35 @@ func TestTokenBucket(t *testing.T) {
 	// the positive.
 	checkExhausted(initialExhausted + (20+90)*time.Millisecond)
 }
+
+func TestWaitCtx(t *testing.T) {
+	var tb TokenBucket
+	tb.Init(1, 100)
+	// Drain the initial tokens.
+	if fulfilled, _ := tb.TryToFulfill(100); !fulfilled {
+		t.Fatalf("could not drain initial tokens")
+	}
+	waitResult := make(chan error, 1)
+	ctx, ctxCancel := context.WithCancel(context.Background())
+	go func() {
+		// This would take 100 seconds to return unless we cancel the context.
+		waitResult <- tb.WaitCtx(ctx, 100)
+	}()
+
+	time.Sleep(10 * time.Millisecond)
+	select {
+	case <-waitResult:
+		t.Fatal("WaitCtx terminated unexpectedly")
+	default:
+	}
+
+	ctxCancel()
+	select {
+	case err := <-waitResult:
+		if err == nil || !strings.Contains(err.Error(), "context canceled") {
```

Comment:
nit: can we replace the string comparison with `errors.Is(err, context.Canceled)`?

Reply:
Done.